### PR TITLE
move to MarkdownSnippets

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,60 +30,84 @@ To Install from the Nuget Package Manager Console
...
+        messagePropertyToIdentifySasUri: "mySasUriProperty");
 sender.RegisterAzureStorageAttachmentPlugin(config);
-```  
+```
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L51-L60)</sup>
+<!-- endsnippet -->
 
 Sending
 
-```c#
```
Comment (SimonCropp) [inline README.md]: is this snippet meant to be the same as the above `AttachmentSending` one?

File: README.md [hunk 2]
```diff
@@ -107,27 +137,37 @@ new AzureStorageAttachmentConfiguration(storageConnectionString).WithSasUri(mess
 
 Default is to convert any body to attachment.
 
-```c#
+<!-- snippet: Configure_criteria_for_message_max_size_identification -->
+```cs
 // messages with body > 200KB should be converted to use attachments
-new AzureStorageAttachmentConfiguration(storageConnectionString, message => message.Body.Length > 200 * 1024);
```
Comment (SimonCropp) [inline README.md]: this didnt compile

File: README.md [hunk 3]
```diff
@@ -24,60 +30,84 @@ To Install from the Nuget Package Manager Console
...
 Receiving only mode (w/o Storage account credentials)
 
-```c#
-// Overide message property used to identify SAS URI
+<!-- snippet: AttachmentReceivingSas -->
+```cs
+// Override message property used to identify SAS URI
 // .RegisterAzureStorageAttachmentPluginForReceivingOnly() is using "$attachment.sas.uri" by default
-var receiver = messageReceiver.RegisterAzureStorageAttachmentPluginForReceivingOnly("mySasUriProperty");
-var msg = await receiver.ReceiveAsync().ConfigureAwait(false);
```
Comment (SimonCropp) [inline README.md]: this didnt compile

File: README.md [hunk 4]
```diff
@@ -107,27 +137,37 @@ new AzureStorageAttachmentConfiguration(storageConnectionString).WithSasUri(mess
...
 var config = new AzureStorageAttachmentConfiguration(provider);
 ```
+<sup>[snippet source](/src/ServiceBus.AttachmentPlugin.Tests/Snippets.cs#L102-L106)</sup>
+<!-- endsnippet -->
 
 ### Configuring plugin using StorageCredentials (Service or Container SAS)
 
-```c#
-var credentials = new StorageCredentials(/*Shared key OR Service SAS OR Container SAS*/);
-var config = new AzureStorageAttachmentConfiguration(credentials);
```
Comment (SimonCropp) [inline README.md]: this didnt compile

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.404-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/22590232) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/eeb63a1abf by @SimonCropp)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.405-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/22590264) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/7c36ce3516 by @SimonCropp)

**Comment by SeanFeldman** [comment]
@SimonCropp I'm confused. All "this didn't compile" snippets are compiling fine. Otherwise the build would fail, but it's [passing](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/22590264). Am I missing something?

**Comment by SimonCropp** [comment]
They were not compiling if u use them as they were previously.

File: README.md (hunk 1 above)
Comment (SeanFeldman) [reply README.md]: Sending using connection string or SAS token looks the same. Could be merged I suppose 🙂

**Comment by SeanFeldman** [comment]
That's weird. Looking at https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/pull/89/files#diff-04c6e90faac2675aa89e2176d2eec7d8R143, for example, I can't see any difference. Though original snippets were copy/paste. With compiled snippets it's guaranteed to work.

Regarding https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/pull/89#discussion_r259563596 I don't care that much for duplication. Will merge and if needed, can be addressed later. Deal?

**Comment by SeanFeldman** [comment]
Thank you @SimonCropp 🙂

**Pull request merged**